fdbserver/IPager.h

```diff
@@ -76,16 +77,69 @@ static const std::vector<std::pair<PagerEvents, PagerEventReasons>> L0PossibleEv
 	{ PagerEvents::PageWrite, PagerEventReasons::MetaData },
 };
 
-// Represents a block of memory in a 4096-byte aligned location held by an Arena.
+enum EncodingType : uint8_t {
+	XXHash64 = 0,
+	// For testing purposes
+	XOREncryption = 1
+};
+
+enum PageType : uint8_t {
+	HeaderPage = 0,
+	BackupHeaderPage = 1,
+	BTreeNode = 2,
+	BTreeSuperNode = 3,
+	QueuePageStandalone = 4,
+	QueuePageInExtent = 5
+};
+
+// Encryption key ID
+typedef uint64_t KeyID;
+
+// Encryption key and secret string
+struct KeyIDWithSecret {
+	KeyID id;
+	Key secret;
+};
+
+// Interface used by pager to get encryption keys by ID when reading pages from disk
+// and by the BTree to get encryption keys to use for new pages
+class IEncryptionKeyProvider {
```

Comment:
This is great! 
Not related to this PR (and we don't have to change anything right now), just to share the current plan; new EncryptRole would implement API to do the following:
1. refresh encryption periodically
2. Lookup encryption key on-demand

Plan is to create an in-memory cache of fetched encryption-keys (time bound) in EncyrptKeyServer, however, also create encryptKey caches in all other processes using the encyrpt-keys such as: TLog/CP (based on scheme we select), SS, BlobManager/Worker etc. Given encyrpt-key-refreshes are done asynchronously between GS <-> FDB and their is no strict requirement about encyrpt-key usages (for new writes), their is big performance gain by maintain encyrpt-key caches in above mentioned processes. 

EncryptKeyServer can broadcast when the encyrpt-key changes (seldom given encrypt-key lifespan is 30 days), optimizing the network traffic due to encrypt-key reasons.

Reply:
Now that I've implemented `EncryptionKeyRef` it did occur to me that TTL could be added to it so if an `IEncryptionKeyProvider` implements a cache it can obey the TTL.

We could maybe implement the client side cache as an `IEncryptionKeyProvider` which wraps/proxies another `IEncryptionKeyProvider` instance which knows how to talk to the `EncryptionKeyServer`.

Or, potentially there are more layers here than needed and we could just make `EncryptionKeyClient` which talks to `EncryptionKeyServer` and handles caching / TTL and implements the interface Redwood is using or something analogous so we can just have the Redwood pager and btree take a reference to the `EncryptionKeyClient` directly.

Whichever interface Redwood ends up using, there will need to be mock implementations of it, such as what `XOREncryptionKeyProvider` is currently, because Redwood has test use cases which run without an FDB cluster (so no encryption key server role) and encryption needs to be usable in those instances.